packages/zoe/src/contractFacet/offerSafety.js

```diff
@@ -18,7 +18,22 @@ const satisfiesInternal = (getAmountMath, giveOrWant = {}, allocation) => {
     }
     const amountMath = getAmountMath(requiredAmount.brand);
     const allocationAmount = allocation[keyword];
-    return amountMath.isGTE(allocationAmount, requiredAmount);
+
+    // if requiredAmount is empty, it is not a useful searchAmount and
+    // we can return early
+    if (amountMath.isEmpty(requiredAmount)) {
+      return amountMath.isGTE(allocationAmount, requiredAmount);
```

Comment:
Once we write down the algebraic requirements on amountMath, I think we'll find that any amount is necessarily GTE the empty amount. If you still want to check it here, I'd prefer an `assert` since a violation would be a bug.

Reply:
This treatment of the empty searchAmount is opposite of `find`'s treatment of the empty searchAmount, which is why you need to break this out as a separate test. For `find`, no amount matches the empty searchAmount. For `isGTEByKeyword`, any amount satisfies the empty amount.